node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,329 @@
+
+type AddressEntry = [string, number];
+
+describe("Auth tests", () => {
+    let cmeCluster: ValkeyCluster;
+    let cmdCluster: ValkeyCluster;
+    let managementClient: BaseClient;
+    let client: BaseClient;
+    beforeAll(async () => {
+        const standaloneAddresses = global.STAND_ALONE_ENDPOINT;
+        const clusterAddresses = global.CLUSTER_ENDPOINTS;
+
+        // Connect to cluster or create a new one based on the parsed addresses
+        cmdCluster = standaloneAddresses
+            ? await ValkeyCluster.initFromExistingCluster(
+                  false,
+                  parseEndpoints(standaloneAddresses),
+                  getServerVersion,
+              )
+            : await ValkeyCluster.createCluster(false, 1, 1, getServerVersion);
+
+        cmeCluster = clusterAddresses
+            ? await ValkeyCluster.initFromExistingCluster(
+                  true,
+                  parseEndpoints(clusterAddresses),
+                  getServerVersion,
+              )
+            : await ValkeyCluster.createCluster(true, 3, 1, getServerVersion);
+
+        // Create appropriate client based on mode
+        const isStandaloneMode = !!standaloneAddresses;
+        const activeCluster = isStandaloneMode ? cmdCluster : cmeCluster;
+        const ClientClass = isStandaloneMode ? GlideClient : GlideClusterClient;
+
+        managementClient = await ClientClass.createClient({
+            addresses: formatAddresses(activeCluster.getAddresses()),
+        });
+    }, 40000);
+
+    const formatAddresses = (
+        addresses: AddressEntry[],
+    ): { host: string; port: number }[] =>
+        addresses.map(([host, port]) => ({ host, port }));
+
+    afterEach(async () => {
+        if (managementClient) {
+            try {
+                await managementClient.customCommand(["AUTH", "new_password"]);
+                await managementClient.configSet({ requirepass: "" });
+                await managementClient.flushall();
+            } catch {
+                // Ignore errors
+            }
+
+            try {
+                await client.updateConnectionPassword("");
+            } catch {
+                // Ignore errors
+            }
+        }
+
+        if (cmdCluster) {
+            await flushAndCloseClient(false, cmdCluster.getAddresses());
+        }
+
+        if (cmeCluster) {
+            await flushAndCloseClient(true, cmeCluster.getAddresses());
+        }
+    });
+
+    afterAll(async () => {
+        if (cmdCluster) {
+            await cmdCluster.close();
+        }
+
+        if (cmeCluster) {
+            await cmeCluster.close();
+        }
+
+        if (managementClient) {
+            managementClient.close();
+        }
+    });
+
+    const runTest = async (
+        test: (client: BaseClient) => Promise<void>,
+        protocol: ProtocolVersion,
+        configOverrides?: Partial<BaseClientConfiguration>,
+    ) => {
+        const isStandaloneMode = configOverrides?.addresses?.length === 1;
```

Comment:
We can run cluster and create config with only 1 address

Reply:
I think this happens when you use `--cluster-endpoints`

Reply:
Yea, its dealt in setup of all tests then above